### DISCUSSION TITLE
Why doesn't `UseLazyQueryTrigger` return a Promise?

### DISCUSSION THREAD

**Discussion by chandlervdw** [reduxjs/redux-toolkit#1151] [Q&A]
Sorry, this is the only place I know to ask questions. I want to be able to use a simple `.then()` as a response to the trigger pulled from the auto-generated useLazyQuery hook but as it stands, [`UseLazyQueryTrigger`](https://deploy-preview-1016--redux-starter-kit-docs.netlify.app/rtk-query/api/created-api/hooks#signature-4) returns `void`. Is there some reasoning behind this?

This is because I only want to fire the `.then()` once upon completion of the first request. If I can't use that, what's the best combination of props from the `UseQueryStateResult` in a `useEffect()` to ensure that I only fire the second request once?

**Comment by phryneas** [comment]
The simple answer: it's complex ;)

We allow every query to only run once concurrently. So if you were to trigger the query twice in succession very quickly, the first trigger would start the request and the second trigger would skip. 
The second query would now either be able to quickly resolve (which would be wrong) or reject (which would be unexpected to users). In the long term, we might add some more internals that the second query would wait for the first to finish and *then* resolve, but that logic simply is not in place.

So, we return `void`.


As for your current problem: What about something like this?

```js
const firstResult = useLazyQuery(...)
const secondResult = useQuery(..., { skip: !firstResult.isFulfilled })
```

**Reply by bandersongit** [reply]
The docs are currently out of date if this is the behavior. https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#uselazyquery

**Reply by phryneas** [reply]
Actually, by now that is actually in.
